### PR TITLE
Add analyzer for substring checks

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Refs https://github.com/xunit/xunit/issues/1260

### CODE REVIEW

File: src/xunit.analyzers/xunit.analyzers.csproj [hunk 1]
```diff
@@ -32,9 +32,11 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="AssertEqualShouldNotBeUsedForBoolLiteralCheck.cs" />
+    <Compile Include="AssertSubstringCheckShouldNotUseBoolCheckCheck.cs" />
```
Comment (marcind) [inline src/xunit.analyzers/xunit.analyzers.csproj]: This seems wrong here.

Comment (ErikSchierboom) [reply src/xunit.analyzers/xunit.analyzers.csproj]: Whoops. I'll fix this tonight.

Comment (ErikSchierboom) [reply src/xunit.analyzers/xunit.analyzers.csproj]: Fixed!

**Review by marcind — APPROVED** [review]

**Comment by marcind** [comment]
Closing/reopening for CLA bot

**Pull request closed without merging**